modules/gdscript/gdscript_parser.cpp

```diff
@@ -4366,11 +4372,8 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 									else
 										first = false;
 
-									current_export.hint_string += E->get().operator String().camelcase_to_underscore(true).capitalize().xml_escape();
-									if (!is_flags) {
-										current_export.hint_string += ":";
-										current_export.hint_string += enum_values[E->get()].operator String().xml_escape();
-									}
```

Comment:
This works when the values of the elements are sequential starting from 0 (0, 1, 2, 3, 4...), but it won't work when the enum has custom values. See #22005 how I had to add the explicit `:val` suffix to the hint string to fix #17581.

Reply:
Oh, so I should re-edit my PR to remove that change then, and just leave the numbers in?

Reply:
Yes, these lines should not be removed:

``` C++
if (!is_flags) {
    current_export.hint_string += ":";
    current_export.hint_string += enum_values[E->get()].operator String().xml_escape();
}
```

Everything else looks good.

Reply:
Just fixed this.